### PR TITLE
Remove constants (`tokenList.json`, `WALLETInitialClaimableRewards.json` and `humanizerInfo.json`) and instead fetch them in async manner from jason.ambire.com

### PR DESCRIPTION
**Pull request by PetromirDev**
Introduces `useConstants` custom hook, responsible for getting `tokenList.json`, `WALLETInitialClaimableRewards.json`, `humanizerInfo.json` app constants and the `AdexToStakingTransfersLogs.json` log.

All synchronous logic for accessing these constants now is changed to async, plus `usePortfolio` and `useClaimableWalletToken` accept `useConstants` as a prop.

Generating `humanizerInfo` and updating all others now is moved to another repo: https://github.com/AmbireTech/ambire-constants.

### CODE REVIEW

File: src/hooks/usePortfolio/usePortfolio.ts [hunk 1]
```diff
@@ -91,13 +95,15 @@ async function supplementTokensDataFromNetwork({
...
 export default function usePortfolio({
+  fetch, 
   currentNetwork,
   account,
   useStorage,
   isVisible,
   useToasts,
   getBalances
 }: UsePortfolioProps): UsePortfolioReturnType {
+  const { constants, isLoading } = useFetchConstants({ fetch })
```
Comment (superKalo) [inline src/hooks/usePortfolio/usePortfolio.ts:106]: I think it's better to pass either the `useFetchConstants` here, either the `constants` and `isLoading` params. Let's discuss this offline.

File: src/hooks/useClaimableWalletToken/useClaimableWalletToken.ts [hunk 2]
```diff
@@ -14,20 +14,29 @@ const supplyControllerAddress = '0xc53af25f831f31ad6256a742b3f0905bc214a430'
...
 
 const useClaimableWalletToken = ({
+  fetch,
   accountId,
   network,
   addRequest,
   totalLifetimeRewards,
   walletUsdPrice
 }: UseClaimableWalletTokenProps): UseClaimableWalletTokenReturnType => {
+  const { isLoading, constants } = useFetchConstants({ fetch })
```
Comment (superKalo) [inline src/hooks/useClaimableWalletToken/useClaimableWalletToken.ts:24]: I think it's better to pass either the `useFetchConstants` here, either the `constants` and `isLoading` params. Let's discuss this offline.

File: src/services/humanizers/AmbireIdentity.ts [hunk 3]
```diff
@@ -2,17 +2,18 @@
...
+import { HumanizerInfoType } from 'hooks/useFetchConstants'
 import accountPresets from '../../constants/accountPresets'
 import privilegesOptions from '../../constants/privilegesOptions'
 
 import { getName } from '../humanReadableTransactions'
 
 const iface = new Interface(require('adex-protocol-eth/abi/Identity5.2'))
 
-const IdentityMapping = {
+const IdentityMapping = (humanizerInfo:HumanizerInfoType) => ({
```
Comment (superKalo) [inline src/services/humanizers/AmbireIdentity.ts:13]: Let's discuss how to approach this.

File: src/services/address/address.ts [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
 const isValidAddress = (address: string) => /^0x[a-fA-F0-9]{40}$/.test(address)
 
-const isKnownTokenOrContract = (address: string) => {
-  const addressToLowerCase = address.toLowerCase()
-  const tokensAddresses = Object.keys(tokens)
-  const contractsAddresses = Object.keys(names)
-  return (
-    tokensAddresses.includes(addressToLowerCase) || contractsAddresses.includes(addressToLowerCase)
-  )
+const isKnownTokenOrContract = (humanizerInfo:HumanizerInfoType, address: string) => {
```
Comment (superKalo) [inline src/services/address/address.ts:5]: Let's discuss how to approach this.

File: src/hooks/useFetchConstants/useFetchConstants.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                }
+              : null
+          )
+          return false
+        })
+      } else {
+        setIsLoading(false)
+      }
+    }
+    fetchConstants()
```
Comment (superKalo) [inline src/hooks/useFetchConstants/useFetchConstants.ts:53]: Let's discuss what happens if this endpoint fails (error handling)

**Review by superKalo — APPROVED** [review]
All right, as far as I can tell - all changes make sense and it looks good to me! 👍 

Approved, but on hold, until the web & mobile integrations are finalized and until we figure out a release plan for this breaking change.

**Review by sonytooo — APPROVED** [review]

**Pull request merged**